src/OpenDebugAD7/Tracepoint.cs

```diff
@@ -0,0 +1,523 @@
+                case "$PNAME":
+                case "$CALLER":
+                case "$CALLSTACK":
+                case "$TICK":
+                    return true;
+                    break;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Method to find the end matching quote.
+        /// </summary>
+        /// <param name="input">string to find quote. Index 0 is the open quote.</param>
+        /// <param name="endIndex">output of where the index of the end quote relative to 'input'</param>
+        /// <returns>true if a end quote was found</returns>
+        private bool FindEndQuote(string input, out int endIndex)
+        {
+            endIndex = -1;
+
+            int index;
+            bool isEscaped = false;
+            for (index = 1; index < input.Length; index++)
+            {
+                char c = input[index];
+
+                if (c == '"' && !isEscaped)
+                {
+                    endIndex = index;
+                    return true;
+                }
+
+                if (isEscaped)
+                {
+                    isEscaped = false;
+                }
+
+                if (c == '\\')
+                {
+                    isEscaped = true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Method to find the interpolated expression.
+        /// E.g. Find the end curl brace. 
+        /// </summary>
+        /// <param name="input">String to find the end brace. Index 0 is '{'</param>
+        /// <param name="endIndex">Index of '}' relative to input.</param>
+        /// <returns>true if an end brace was found.</returns>
+        private bool FindInterpolatedExpression(string input, out int endIndex)
+        {
+            endIndex = -1;
+            int index = 0;
+            StringBuilder buffer = new StringBuilder();
+            int nested = 0;
+            bool isEscaped = false;
+            while (++index < input.Length)
+            {
+                char c = input[index];
+                if (isEscaped)
+                {
+                    buffer.Append(c);
+                    isEscaped = false;
+                }
+                else
+                {
+                    if (c == '\\')
+                    {
+                        isEscaped = true;
+                    }
+                    else if (c == '{')
+                    {
+                        nested++;
+                    }
+                    else if (c == '}')
+                    {
+                        if (nested > 0)
+                        {
+                            nested--;
+                        }
+                        else
+                        {
+                            endIndex = index;
+                            return true;
+                        }
+                    }
+                    else if (c == '"')
```

Comment:
>c == '"') [](start = 29, length = 9)

Does VS special case quoted strings also? This feels like it isn't really necessary -- if I evaluated `"{myVarWithValue1}"`, getting my `"1"` seems okay to me. But if VS does it also, I don't have a problem doing the same.

Reply:
Sorry, wrong spot.

---
In reply to: [444434372](https://github.com/microsoft/MIEngine/pull/1013#discussion_r444434372) [](ancestors = 444434372)